File: module/storage/db_stats.go
package main

import (
	"github.com/pantopic/wazero-lmdb/lmdb-go"
)

type dbStatsImpl struct {
	db
}

func (db dbStatsImpl) init(txn *lmdb.Txn) {
	db.open(txn)
}

File: module-zig/storage-kv/db_stats.zig
//! Mirrors module/storage-kv/db_stats.go

const lmdb = @import("lmdb");

const Db = @import("db.zig").Db;

pub const DbStats = struct {
    db: Db,

    pub fn init(self: DbStats, txn: lmdb.Txn) void {
        self.db.open(txn);
    }
};
